base/libuv.jl

```diff
@@ -66,7 +66,16 @@ uverrorname(err::UVError) = uverrorname(err.code)
 
 uv_error(prefix::Symbol, c::Integer) = uv_error(string(prefix),c)
 uv_error(prefix::AbstractString, c::Integer) = c < 0 ? throw(UVError(prefix,c)) : nothing
-show(io::IO, e::UVError) = print(io, e.prefix*": "*struverror(e)*" ("*uverrorname(e)*")")
+
+function showerror(io::IO, e::UVError)
+    print(io, e.prefix, ": ", struverror(e), " (", uverrorname(e), ")")
+    if e.code == UV_ENOENT && '~' in e.prefix
```

Comment:
Some changes will be needed now:
```suggestion
    print(io, e.msg, ": ", struverror(e.code), " (", uverrorname(e.code), ")")
    if e.code == UV_ENOENT && '~' in e.msg
```

some testing options:
```julia
julia> using Test

julia> @test_throws "Many shells" cd("~")

julia> @test occursin("Many shells", sprint(showerror, Base.IOError("~", Base.UV_ENOENT)))
```